### PR TITLE
gNMI-1.10: update sw version check according to the new README

### PR DESCRIPTION
**Pull request by LimeHat**
This PR implements sw version check according to the updates specified in #1313 

---
<sup>
This code is a Contribution to the OpenConfig Feature Profiles project ("Work") made under the Google Software Grant and Corporate Contributor License Agreement ("CLA") and governed by the Apache License 2.0. No other rights or licenses in or to any of Nokia’s intellectual property are granted for any other purpose. This code is provided on an "as is" basis without any warranties of any kind.</sup>

### CODE REVIEW

**Comment by xw-g** [comment]
https://b.corp.google.com/issues/274004168#comment12 is the internal bug.

**Comment by jjing1122** [comment]
File internal bug b/278733676 for presubmit process.

**Comment by sgirishj** [comment]
In order to verify that the CONTROLLER_CARD(s) have a valid software-version and to fetch its value, why can't we use the below leaf under the CONTROLLER_CARD component(s) directly?
state/software-version

**Comment by LimeHat** [comment]
@sgirishj that was the initial proposal in #1279, but as the result of this conversation the requirement was clarified/changed in #1313 

@xw-g can elaborate on the reasoning.

**Comment by sgirishj** [comment]
> @sgirishj that was the initial proposal in #1279, but as the result of this conversation the requirement was clarified/changed in #1313
> 
> @xw-g can elaborate on the reasoning.

Ok, thanks @LimeHat!

File: internal/components/components.go
Comment (mojiiba) [inline internal/components/components.go:67]: these logging is not very helpful and fill up the test log, can u please remove that

File: feature/gnmi/otg_tests/telemetry_basic_check_test/telemetry_basic_check_test.go
Comment (LimeHat) [reply feature/gnmi/otg_tests/telemetry_basic_check_test/telemetry_basic_check_test.go:472]: I believe a deviation would require a separate approval so I'd propose to do it via a different PR if you don't mind

Comment (vosipchu) [reply feature/gnmi/otg_tests/telemetry_basic_check_test/telemetry_basic_check_test.go:472]: let's have a deviation not to fail a vendor that is passing this right now. Approvals will be fast. 
@mojiiba can you please propose what needs to be added? Thanks

cc @xw-g

Comment (jjing1122) [reply feature/gnmi/otg_tests/telemetry_basic_check_test/telemetry_basic_check_test.go:472]: do we all agree on opening a different PR to approve the deviation and hold this PR until that one is merged?

Comment (LimeHat) [reply feature/gnmi/otg_tests/telemetry_basic_check_test/telemetry_basic_check_test.go:472]: All right I added deviation `SwVersionUnsupported`

@vosipchu I'm assuming you'll take it from here?

**Review by mojiiba — APPROVED** [review]

**Review by jjing1122 — APPROVED** [review]

**Pull request merged**
